### DISCUSSION TITLE
set combobox open menu without using useComboboxPrimitive

### DISCUSSION THREAD

**Discussion by azhengtwilio** [twilio-labs/paste#2350] [Q&A] [answered]
Hi there,
Is there a way where I can set the [Combobox](https://paste.twilio.design/components/combobox/#about-combobox) dropdown menu to open based on certain conditions without me having to use useComboboxPrimitive's `openMenu`?  I only see `initialIsOpen` in the Combobox doc.

I am trying to implement an Anchor link in a toast that will open the dropdown when clicked on
![Screen Shot 2022-04-07 at 3 29 24 PM](https://user-images.githubusercontent.com/83669618/162330082-5ed16390-22c9-4281-a6bc-0cec2b04a7bb.png)

**Comment by TheSisb** [comment, marked as answer]
Hi @azhengtwilio,

> Is there a way where I can set the [Combobox](https://paste.twilio.design/components/combobox/#about-combobox) dropdown menu to open based on certain conditions without me having to use useComboboxPrimitive's openMenu? I only see initialIsOpen in the Combobox doc.

Opening the Combobox menu programmatically is possible with the [state hook `useCombobox`](https://paste.twilio.design/components/combobox/#usecombobox-state-hook) from the non-primitive component.  [Here's an example of how it would work](https://codesandbox.io/s/red-thunder-n6w5wt?file=/src/index.tsx). This is the only way to tap into that functionality.


> I am trying to implement an Anchor link in a toast that will open the dropdown when clicked on

Please do be aware that [Anchors are for hyperlinking to another URL](https://paste.twilio.design/components/anchor#about-anchors) and that [Buttons trigger an action](https://paste.twilio.design/components/button#about-buttons).  In this instance, you likely want a Button. Please consider the [link style button](https://paste.twilio.design/components/button#link-style-button) as it is more accessible for screen reader users.

**Reply by azhengtwilio** [reply]
Thank you, I'll try this out!

**Reply by azhengtwilio** [reply]
@TheSisb is there a way to pass openMenu to the parent component of the Combobox? Here's a PR https://code.hq.twilio.com/wireless/iot-console/pull/412/files

1) I tried passing down an extra param into the component with the combobox implemented to trigger openMenu with useEffect by setting it to `true` when user clicks on it. However since I'm using useEffect to check if the param changes, it won't open the menu the second time being clicked because the param value would true the whole time.
2) I tried writing the component with the combobox into a hook that return an object. The object passes back openMenu and the render of the component but it seems to get into an infinite loop when I do that

**Reply by TheSisb** [reply]
In React you can't really pass things up the tree, but [you can pass them down](https://www.educative.io/edpresso/what-is-unidirectional-data-flow-in-react). If you import and define `useCombobox` in the parent, you can then pass down the state as prop to your combobox component.
